edx_rbac/utils.py

```diff
@@ -203,7 +207,15 @@ def get_decoded_jwt(request):
     Decodes the request's JWT from either cookies or auth payload and returns it.
     Defaults to an empty dictionary.
     """
-    decoded_jwt = get_decoded_jwt_from_cookie(request) or get_decoded_jwt_from_auth(request)
+    try:
+        decoded_jwt_from_cookie = get_decoded_jwt_from_cookie(request)
+    except InvalidTokenError as exc:
+        if getattr(settings, 'IGNORE_INVALID_TOKEN_ERROR', False):
```

Comment:
I think part of the issue is that this exception is blocking from trying the second auth option: `get_decoded_jwt_from_auth`. It feels like we may want to always suppress this exception until the other call gets a chance to work or fail. 

If `get_decoded_jwt_from_auth` succeeds, I'm not sure that anyone would care about the original failure. If it fails, you'd need to decide whether you want to provide details about the earlier error in logs. But this might avoid the use of a new setting, and make this work more like it seems it was designed to work in all cases.

Reply:
Ah yeah, that's a good point.  So maybe something like this?
```python
```

Reply:
(excs_from_cookie doesn't really need to be a list...)

Reply:
The cases where we are burying the exception is a little misleading here. Maybe the code would be a bit more clear if we worked with `get_decoded_jwt_from_auth` first? This might require something like `IGNORE_INVALID_TOKEN_ERROR` to come back?

That said, I'm not clear if `IGNORE_INVALID_TOKEN_ERROR` is just an observability issue, and what would be the best default behavior? Would any auth errors have been handled appropriately, and could one argue that this method is only interested in successful JWTs? Is this blocking alternative checks for access?

You could always add `IGNORE_INVALID_TOKEN_ERROR` with it defaulting to expected behavior, but with the ability to enable. My head isn't in the code as much as you, so these are things to consider. Note that **IF** you went down this path, then bug-fix might make more sense (if that is what you think this is), rather than a backward-incompatible major version bump. If you don't think this is a bug fix, then maybe your original code (after switching the order of attempts) would be best.

If you provide the log message:
1. Should this be an info, rather than a warn? 
2. Should the message give some indication of the context?

Reply:
OK, this is good feedback, I implemented it as follows:
1. `get_decoded_jwt_from_auth()` is now evaluated first.
2. For ease of risk mitigation, I switched back to raising `InvalidTokenException` by default if `get_decoded_jwt_from_cookie()` raises it.
3. I added back a setting `IGNORE_INVALID_JWT_COOKIE` that, when true, will cause `get_decoded_jwt()` to _not_ raise.
4. A `logger.info()` is called whenever we encounter `InvalidTokenException` in `get_decoded_jwt()` - this happens regardless of the state of the setting above.

To me, this approach provided the best mix of readability, observability, and backwards compatibility.